### PR TITLE
action.yml ready for testing

### PR DESCRIPTION
**Pull request by callmestu**
# Contributor Comments
This branch creates a simple Github action to checkout a docker image and run the default command "terraform validate."

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, align your PR
  title with [this documentation](https://help.github.com/en/articles/closing-issues-using-keywords)

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: 'Easy Infra[structure as Code]'
+description: 'A docker container to simplify and secure the use of Infrastructure as Code (IaC)'
+inputs:
+  command:
+    description: 'Command to pass to and execute within the Docker container.'
+    required: true
+    default: 'terraform validate'
+runs:
+  using: 'docker'
+  image: 'docker://docker.io/seiso:easy_infra:latest'
```
Comment (JonZeolla) [inline action.yml:10]: I think the double : is a typo

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: 'Easy Infra[structure as Code]'
+description: 'A docker container to simplify and secure the use of Infrastructure as Code (IaC)'
+inputs:
+  command:
+    description: 'Command to pass to and execute within the Docker container.'
```
Comment (JonZeolla) [inline action.yml:5]:
```suggestion
    description: 'Command to execute within the Docker container.'
```

File: action.yml (hunk 1 above)
Comment (callmestu) [reply action.yml]: Thanks, fixed.

File: action.yml (hunk 2 above)
Comment (callmestu) [reply action.yml:5]: Cool, concision.

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: 'Easy Infra[structure as Code]'
```
Comment (JonZeolla) [inline action.yml:1]: Can you point me to how you've tested this?  Thanks!

**Review by JonZeolla — CHANGES_REQUESTED** [review]
Thoughts inline; thanks!

File: README.md [hunk 4]
```diff
@@ -66,3 +66,7 @@ If you'd like to enable debug logs at runtime, pass an environment variable of `
 ```bash
 docker run -e LOG_LEVEL=DEBUG -v $(pwd):/iac seiso/easy_infra terraform validate
 ```
+
+### Automation with GitHub Actions
+
+A basic GitHub action defaults to passing command `terraform validate` to the container. This command is optional to allow using non-Terraform actions as well.
```
Comment (JonZeolla) [inline README.md:72]: We should document how someone would actually use this.  What is the relevant config they should add, and where?  Consider looking at the documentation in [the goat](https://github.com/seisollc/goat) for inspiration.

File: docs/index.rst [hunk 5]
```diff
@@ -4,7 +4,9 @@
    contain the root `toctree` directive.
 
 
-.. mdinclude:: ../README.md
+.. mdinclude:: 
+   ../README.md
+   backlog.md
```
[the remark this answers is not in the document]
Comment (callmestu) [reply docs/index.rst:9]: Reverted.

File: action.yml (hunk 3 above)
Comment (callmestu) [reply action.yml:2]: Thanks for pointing that out. I added this action to another repo I invited you to - can you have a look at this action and let me know if I need to add chown/chmod or similar:
https://github.com/callmestu/Logging-AWS-AZ-Terraform/runs/5233640873

**Comment by callmestu** [comment]
This task is replaced by a new, larger project/repo that will add more functionality: Repo for easy_infra/SAST github action.

Cancelling the PR related to this task now.

**Pull request closed without merging**